actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
Btw just in case of error so @sgrif is right on in normal case but this is not important than the possible confusion this implementation can lead.